beetsplug/convert.py

```diff
@@ -48,6 +48,8 @@ def replace_ext(path, ext):
 
     The new extension must not contain a leading dot.
     """
+    assert isinstance(path, bytes)
+    assert isinstance(ext, bytes)
```

Comment:
Asserting things like this isn't a good way to handle it. It's the equivalent of throwing up your hands and saying 'screw it'. Is there a way to test for the problem and do something? The best solution would be to make sure that something that isn't bytes isn't passed in. Do you have reason to believe that something else will be passed in?

Reply:
I think it's just a matter of another (bad ?) developer habit. Since I sometimes struggles to find what kind of data type is the input of a function, and since passing another data type will lead to errors, I'm used to perform sanity check using assertion to catch developers errors (not user errors). 

That being say, I did not had strong reasons to do so here nor to believe that something else would be passed in, except a developer error. If it is not appreciated by the coding habits / rules of Beets, for sure I will remove this.